Bullwinkle.class.nut

```diff
@@ -162,11 +166,20 @@ class Bullwinkle {
         if (_watchdogTimer == null) _watchdog();
 
         // Increment the tries
-        if (message.type == BULLWINKLE_MESSAGE_TYPE.SEND) {
-        	message.tries++;
+        if (message.type == BULLWINKLE_MESSAGE_TYPE.SEND && message.id in _packages) {
+		        _packages[message.id]._tries++;
+        }
+
+        if(imp.getmemoryfree() > _settings.lowMemoryThreshold && (_isAgent() || server.isconnected())){
```

Comment:
Forgive my bias towards wanting the feature :) 

That seems reasonable to me to preserve existing functionality.